src/engine/bufferscalers/rubberbandwrapper.cpp

```diff
@@ -0,0 +1,174 @@
+}
+size_t RubberBandWrapper::getLatency() const {
+    VERIFY_OR_DEBUG_ASSERT(isValid()) {
+        return -1;
+    }
+    return m_pInstances[0]->getLatency();
+}
+double RubberBandWrapper::getPitchScale() const {
+    VERIFY_OR_DEBUG_ASSERT(isValid()) {
+        return -1;
+    }
+    return m_pInstances[0]->getPitchScale();
+}
+// See
+// https://github.com/breakfastquay/rubberband/commit/72654b04ea4f0707e214377515119e933efbdd6c
+// for how these two functions were implemented within librubberband itself
+size_t RubberBandWrapper::getPreferredStartPad() const {
+    VERIFY_OR_DEBUG_ASSERT(isValid()) {
+        return -1;
+    }
+#if RUBBERBANDV3
+    return m_pInstances[0]->getPreferredStartPad();
+#else
+    // `getPreferredStartPad()` returns `window_size / 2`, while with
+    // `getLatency()` both time stretching engines return `window_size / 2 /
+    // pitch_scale`
+    return static_cast<size_t>(std::ceil(
+            m_pInstances[0]->getLatency() * m_pInstances[0]->getPitchScale()));
+#endif
+}
+size_t RubberBandWrapper::getStartDelay() const {
+    VERIFY_OR_DEBUG_ASSERT(isValid()) {
+        return -1;
+    }
+#if RUBBERBANDV3
+    return m_pInstances[0]->getStartDelay();
+#else
+    // In newer Rubber Band versions `getLatency()` is a deprecated alias for
+    // `getStartDelay()`, so they should behave the same. In the commit linked
+    // above the behavior was different for the R3 stretcher, but that was only
+    // during the initial betas of Rubberband 3.0 so we shouldn't have to worry
+    // about that.
+    return m_pInstances[0]->getLatency();
+#endif
+}
+void RubberBandWrapper::process(const float* const* input, size_t samples, bool isFinal) {
+    if (m_pInstances.size() == 1) {
+        return m_pInstances[0]->process(input, samples, isFinal);
+    } else {
+        RubberBandWorkerPool* pPool = RubberBandWorkerPool::instance();
+        for (auto& pInstance : m_pInstances) {
+            pInstance->set(input, samples, isFinal);
+            if (!pPool->tryStart(pInstance.get())) {
+                pInstance->run();
+            }
+            input += pPool->channelPerWorker();
+        }
+        for (auto& pInstance : m_pInstances) {
+            pInstance->waitReady();
+        }
+    }
+}
+void RubberBandWrapper::reset() {
+    for (auto& stretcher : m_pInstances) {
+        stretcher->reset();
+    }
+}
+void RubberBandWrapper::clear() {
+    m_pInstances.clear();
+}
+void RubberBandWrapper::setup(mixxx::audio::SampleRate sampleRate,
+        mixxx::audio::ChannelCount chCount,
+        const RubberBandStretcher::Options& opt) {
+    // The instance should have been cleared, or not set before
+    VERIFY_OR_DEBUG_ASSERT(m_pInstances.size() == 0) {
+        m_pInstances.clear();
+    };
+
+    auto channelPerWorker = getChannelPerWorker();
+    qDebug() << "RubberBandWrapper::setup" << channelPerWorker;
+    VERIFY_OR_DEBUG_ASSERT(0 == chCount % channelPerWorker) {
+        m_pInstances.emplace_back(
```

Comment:
The strategy is - since we can not divide the stretching in an evenly distributed number of channels, we fallback to the single instance / not multithreaded implementation. TLDR stretgy is (audio) quality over performance (my understanding is that RB with different channel is likely to produce different result)

Reply:
Ah, I understand. In the unlikely case something got messed up we only create a stereo instance.